packages/core/src/Modules/Cashier/Containers/payment-agent-transfer.jsx

```diff
@@ -9,42 +9,51 @@ import PaymentAgentTransferForm from '../Components/Form/payment-agent-transfer-
 import PaymentAgentTransferReceipt from '../Components/Receipt/payment-agent-transfer-receipt.jsx';
 import Loading from '../../../templates/_common/components/loading.jsx';
 
-class PaymentAgentTransfer extends React.Component {
-    componentDidMount() {
-        this.props.setActiveTab(this.props.container);
-        if (!this.props.is_virtual) {
-            this.props.onMount();
+const PaymentAgentTransfer = ({
+    balance,
+    container,
+    error,
+    is_loading,
+    is_transfer_successful,
+    is_try_transfer_successful,
+    is_virtual,
+    onMount,
+    onUnMount,
+    setActiveTab,
+}) => {
+    React.useEffect(() => {
+        setActiveTab(container);
+        if (!is_virtual) {
```

Comment:
@abdalla-binary shouldn't this has a dependency with `is_virtual` ?

Reply:
Good catch 👍🏻